### ISSUE TITLE
Special characters and foreign keyboard not displayed correctly

### ISSUE THREAD

**Issue by jaimeeee** [zeit/hyper#29]
So I use Spanish ISO keyboard (like this one: http://g01.a.alicdn.com/kf/HTB1iOQ8HVXXXXaeXpXXq6xXFXXXd/Spanish-Euro-Keyboard-Cover-Protector-for-MacBook-Air-11-11-6-Inch-European-ISO-Keyboard-Spain.jpg_640x640.jpg), and it is impossible to type some characters.

For example, "special" characters like ¿¡· are displayed like this: <00c2><00bf><00c2><00a1><00c2><00b7>, same with the ñ.

And other keyboard combinations are just impossible to type, my biggest issue is **~**, since I need to type _Alt + ñ_ but it just doesn't appear. Same issue with any letter with an accent (á, ö, â, etc).

I'm using zsh, if it matters.

**Comment by rauchg** [comment]
@jaimeeee can you please test with 0.2.1? Thanks for taking the time to report this.

**Comment by jaimeeee** [comment]
Still happening on 0.2.1, I get a "ding" sound when I press Alt + ñ

**Comment by johanbrook** [comment]
I can also reproduce for <kbd>alt+n</kbd> (in order to do `ñ`). Getting the ding sound on 0.2.1.

Also, I cannot produce a tilde (`~`) on my (Swedish) keyboard by doing an <kbd>alt</kbd> key combo. No ding sound is played and cursor stays put.

**Comment by rauchg** [comment]
Please retry with `0.3.1`.

**Comment by jaimeeee** [comment]
Now on 0.3.1 and sadly it is still happening, nothing changed imo.

**Comment by michalkvasnicak** [comment]
Have same problem in version `0.5.0`. Using slovak keyboard in order to product `~` pressing <kbd>alt</kbd>+<kbd>ť</kbd> is only doing ding sound.

**Comment by mxstbr** [comment]
Still happening for me (newest version), am on a german keyboard and can't type `@`.

**Comment by bendc** [comment]
Lots of crucial keys like `^` are actually not working for me either (Belgian AZERTY keyboard).

**Comment by LoicMahieu** [comment]
Same than @bendc here. Belgian keyboard and caracters like ``` ` ``` , `^`, `~` are not working.

**Comment by sullenor** [comment]
Looks like I have the same issue in `0.6.0` version with cyrillic symbols.

**Comment by apacala** [comment]
Same. `£` gets displayed as `<00c2><00a3>`.

**Comment by jdesboeufs** [comment]
Same. Unable to produce a tilde on a French keyboard `alt+n`.

**Comment by julientaq** [comment]
same thing using a azerty ISO turned as a qwerty one. No " or ' .

**Comment by rgcr** [comment]
Hello everyone, I have the same issue, my keyboard layout is `Spanish-ISO`  these characters don't work  ^  ~ `  that's annoying because I can't use hyperterm as my default terminal :(

**Comment by luisherranz** [comment]
@rgcr for `cd ~` you can use `cd $HOME` while this get fixed.

**Comment by rgcr** [comment]
Hello @luisherranz, you are right but I can't edit files properly, those keys don't work even in `vim`. Also it's annoying to have to type `$HOME` instead a simple character

**Comment by luisherranz** [comment]
Yeah, if you are using `vi` I suppose you're screwed.

**Comment by mratmartinez** [comment]
Using Spanish keyboard.  

I can't write tildes. Like this: á é í ó ú or even ñ.  
It's showed as:  
> M-CM--.

I'm using UTF-8 keyboard.

**Comment by jccguimaraes** [comment]
Also have this problem, QUERTY, no `~^´`, etc...
This would be really nice to have properly working. Thanks

**Comment by pedrorijo91** [comment]
still happening on `0.8.2`

**Comment by eliorivero** [comment]
Still experiencing this bug on HyperTerm 0.8.3 in macOS 10.11.6 using an english keyboard with `Español - ISO` layout.
Pressing `alt+;` produces `~` in Terminal but produces nothing in HyperTerm.

**Comment by luisherranz** [comment]
:pray:

**Comment by luisherranz** [comment]
:pray:

**Comment by juanmnl** [comment]
Using Version 0.8.3 (0.8.3.873) - Still can't type the tilde ~ into hyperterm. Tried to paste that char from the _keyboard viewer_ in macOS but no luck either.

**Comment by juanmnl** [comment]
ok, so apparently the tilde works on a spanish keyboard, it just doesnt show when typed.
<img width="283" alt="screen shot 2016-11-04 at 7 50 11 pm" src="https://cloud.githubusercontent.com/assets/699945/20026385/ee2c633e-a2c7-11e6-8192-0b8161e7b84e.png">

**Comment by rgcr** [comment]
@juanmnl I think that's not true, `cd` works because if you don't type anything you are moved to your home directory, try open a file typing `~` in your home with vim and you will see that it does not work.

**Comment by zone-live** [comment]
Using version 0.8.3 and Still no ~support, I can copy and paste it, but can't type it.
Already tried a bunch of keys combinations but no luck. I'm using the mac keyboard, QWERTY.

**Comment by matheuss** [comment]
@zone-live this was fixed via #1006 and it'll be released very soon 😄

**Comment by Pawamoy** [comment]
I have the 1.0.0 version and still cannot type an `é` :cry:

**Comment by DanielRamosAcosta** [comment]
I'm in 1.3.3 in Ubuntu 17.04 and I still have this issue.

**Comment by rgcr** [comment]
I have the version **1.3.3** and the issue still persists.

Keyboard layout `Spanish-ISO`

``` 
$ locale                                                                                                                                                                           LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=
```

**Comment by DanielRamosAcosta** [comment]
Same here @rgcr , which OS are you using? This happens to me on Linux but not in Windows...

**Comment by rgcr** [comment]
```
ProductName:    Mac OS X
ProductVersion: 10.11.6
```

**Comment by facosta0787** [comment]
I'm in 1.3.3 in Ubuntu Gnome 17.04 and I still have this issue.

**Comment by DanielRamosAcosta** [comment]
Problem persists in 1.4.3

**Comment by chabou** [comment]
@rgcr @DanielRamosAcosta @facosta0787 
Please follow this issue (with a workaround) https://github.com/zeit/hyper/issues/607
Your issue is not exactly this one.

**Comment by DanielRamosAcosta** [comment]
@chabou exactly! that was the problem, thanks you so much!

**Comment by StephaneDci** [comment]
Hello, 

Same problem on azerty WSL french keyboard : AltGr does not execute correctly
exemple : AltGr<2>space should prints   ~
Actually it prints    é~

WSL windows 10 v1809
`
» locale 
LANG=fr_FR.UTF-8
LANGUAGE=
LC_CTYPE=fr_FR.UTF-8     
LC_NUMERIC="fr_FR.UTF-8" 
LC_TIME="fr_FR.UTF-8"    
LC_COLLATE="fr_FR.UTF-8" 
LC_MONETARY="fr_FR.UTF-8"       
LC_MESSAGES="fr_FR.UTF-8"       
LC_PAPER="fr_FR.UTF-8"
LC_NAME="fr_FR.UTF-8"
LC_ADDRESS="fr_FR.UTF-8"        
LC_TELEPHONE="fr_FR.UTF-8"      
LC_MEASUREMENT="fr_FR.UTF-8"    
LC_IDENTIFICATION="fr_FR.UTF-8" 
LC_ALL=
`

**Comment by Asmoddym** [comment]
exact same problem here

**Comment by Sryther** [comment]
@StephaneDci I have the same issue with a french keyboard. In WSL `AltGr+~` works perfectly but in Hyper it does `é~`. And indeed `Ctrl+Alt+~` is working great!
The difference is my locale: `en_US.UTF-8`.

I tried many things but nothing fixed that.

**Comment by tcardonne** [comment]
Same behavior with a french keyboard.
`AltGr+é` then space prints `é~`.
If I do the combo 2 times : `éé~`.~
And only once without hitting space after, just `é`.

`Ctrl+Alt+é` works as expected, but it requires some gymnastics with my fingers.

This doesn't seem related to any plugin, I tried removing the material theme and the issue persists.
I'm using latest 3.0.2 (stable).

**Comment by hakankaraduman** [comment]
any solution is planned to this? I use a turkish q layout keyboard. I get the correct ~and `characters on any other application. But on the hyper it writes ü~ instead of only ~ and writes ,` instead of only ` character.

**Comment by mocrsteel** [comment]
I'm seeing the same issue with `AltGr + é` followed by `SPACE` on a French keyboard, which also produces `é~` instead of `~`

Some Google'ing gave me the following hit. You never know if it helps:
This issue seems to pop up in different programs too: a VS Code issue:  [https://github.com/microsoft/vscode/issues/59438](url)
Xterm issue (referenced in VS Code issue) [https://github.com/xtermjs/xterm.js/issues/2661](url)
